bin/prepare_feedback.py

```diff
@@ -0,0 +1,173 @@
+    for topic in new_topics:
+        df.topic = (df
+            .topic
+            .replace(to_replace=topic, value=new_topics[topic]))
+    # rename some tutorials
+    for tuto in new_tutorial_name:
+        df.loc[df.tutorial == tuto, 'tutorial'] = new_tutorial_name[tuto]
+    return df
+
+
+def extract_topic_tutorial_name(df):
+    '''Extract topic from tutorial name
+
+    :param df: dataframe
+    '''
+    df = df.copy()
+    new = df['tutorial_topic'].str[::-1].str.split('(', n = 1, expand = True)
+    df["tutorial"]= new[1].str[::-1].str[:-1]
+    df["topic"]= new[0].str[::-1].str[:-1]
+    return df
+
+
+def prepare_feedback(url, out_file):
+    '''Get and prepare feedback CSV file
+
+    :param url: URL to Google sheet with feedback answers
+    :param out_file: Path to output file
+    '''
+    df = (pd.read_csv(url, sep='\t')
+        # rename column
+        .rename(columns = {'Timestamp': 'timestamp',
+                            'How much did you like this tutorial?': 'note',
+                            'What did you like?': 'pro',
+                            'What could be improved?': 'con',
+                            'Tutorial': 'tutorial_topic',
+                            'Your feedback is always anonymous. Also make it confidential (only visible to admins)?': 'anonymous'})
+        # extract topic from tutorial name
+        .pipe(extract_topic_tutorial_name)
+        # remove rows with NaN on note, pro and con
+        .dropna(subset=['note', 'pro', 'con'], how='all')
+        # replace NaN in note by 0
+        .fillna(value={'note': 0})
+        # fill other NaN by empty string
+        .fillna('')
+        # format 
+        .assign(
+            #note to integer
+            note=lambda x: x['note'].astype(int),
+            # format pro and con to string
+            pro=lambda x: x['pro'].astype(str),
+            con=lambda x: x['con'].astype(str),
+            # extract month and date
+            timestamp=lambda x: pd.to_datetime(x['timestamp'], dayfirst=True))
+        .assign(
+            month=lambda x: x['timestamp'].dt.strftime("%Y-%m"),
+            date=lambda x: x['timestamp'].dt.strftime("%Y-%m-%d"))
+        # change topic for some tutorials
+        .pipe(fix_tutorial_info)
+        # remove extra columns
+        .drop(columns=["tutorial_topic", "timestamp"])
+        # remove some rows 
+        .query('topic != "TESTIN"')
+        .query('topic != "tes"')
+        .query('topic != ""')
+        # remove tutorials
+        .query('tutorial != "RNA-seq counts to genes and pathways"')
+        .query('tutorial != "Galaxy Tool Management"')
+        .query('tutorial != "Recording Job Metrics"')
+        #
+        .query('pro != "TESTING"')
+        .to_csv(out_file))
+
+
+if __name__ == '__main__':
+    url = 'https://docs.google.com/spreadsheets/d/1NfZhi5Jav7kl9zFCkeb7rIC2F8xW1isruv1TeO4WpNI/export?format=tsv'
+    out_file = Path('metadata') / Path('feedback.csv')
```

Comment:
should we make this TSV formatted? it'd be a bit safer for others to work with I guess. (Or JSON/yaml?)

Reply:
I am not sure if Jekyll can read TSV file :smile: